src/hooks/use-queries.ts

```diff
@@ -174,6 +180,81 @@ export const useEmployee = (employeeId: string, enabled = true) => {
   });
 };
 
+export const useTimeEntries = (requestParams: ListEmployeeTimeEntriesRequest, salaryGroup: string, enabled = true) => {
+  const { timeEntriesApi } = useApi();
+
+  if (!salaryGroup) {
+    requestParams.start = DateTime.now().startOf("month").toJSDate();
+  }
+
+  // Testing the getWorkingPeriodDates function
+  console.log("WORKING PERIOD DATES", getWorkingPeriodDates(SalaryGroup.Office, DateTime.now().set({month: 3, day: 17}).toJSDate()));
+
+  return useQuery({
+    queryKey: [QUERY_KEYS.TIME_ENTRIES, requestParams],
+    enabled: enabled,
+    queryFn: async () => {
+      const [timeEntries, headers] = await timeEntriesApi.listEmployeeTimeEntriesWithHeaders(requestParams);
+      const totalResults = getTotalResultsFromHeaders(headers);
+
+      return { timeEntries, totalResults };
+    },
+  });
+}
+
+/**
+ * Gets working period start and end datetime according to salary group and selected date
+ * 
+ * @param salaryGroup Salary group of the employee
+ * @param selectedDate Selected date
+ * @returns Start and end date of the working period
+ */
+export const getWorkingPeriodDates = (salaryGroup: string, selectedDate: Date) => {
+  const selectedDateTime = DateTime.fromJSDate(selectedDate);
+
+  if (isOfficeOrTerminalGroup(salaryGroup)) {
+    return getOfficeOrTerminalPeriod(selectedDateTime);
+  }
+    return getNonOfficePeriod(selectedDateTime);
+};
+
+const isOfficeOrTerminalGroup = (salaryGroup: string) => {
+  return salaryGroup === SalaryGroup.Office || salaryGroup === SalaryGroup.Terminal;
+};
+
+const getOfficeOrTerminalPeriod = (selectedDateTime: DateTime) => {
+  const midMonth = 16;
+
+  if (selectedDateTime.day < midMonth) {
+    const start = selectedDateTime.startOf("month").toJSDate();
+    const end = selectedDateTime.set({ day: 15 }).endOf("day").toJSDate();
+    return { start, end };
+  }
+    const start = selectedDateTime.set({ day: 16 }).startOf("day").toJSDate();
+    const end = selectedDateTime.endOf("month").endOf("day").toJSDate();
+    return { start, end };
+};
+
+const getNonOfficePeriod = (selectedDateTime: DateTime) => {
+  const fullWeeksFromStartDate = calculateFullWeeksFromStartDate(selectedDateTime);
+  const isEvenWeek = fullWeeksFromStartDate % 2 === 0;
+
+  if (isEvenWeek) {
+    return calculatePeriod(fullWeeksFromStartDate);
+  }
+    return calculatePeriod(fullWeeksFromStartDate - 1);
```

Comment:
This can be shortened like this, as the remainder is always 0 or 1 when dividing a number by 2 (if rounded up, otherwise 1 week would be wrong).

```suggestion
  const remainderRoundedUp = Math.ceil(fullWeeksFromStartDate % 2);
  return calculatePeriod(fullWeeksFromStartDate - remainderRoundedUp);
```